doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+and add/remove them on _every_ Terminal launch. To make this functionality work
+appropriately, we'll need to introduce a constraint on dynamic profiles.
+
+**For any dynamic profiles, they must be able to be generated using a stable
+GUID**. For example, any time we try adding the "Ubuntu" profile, we must be
+able to generate the same GUID every time. This way, when a dynamic profile
+generator runs, it can check if that profile source already has a profile
+associated with it, and do nothing (as to not create many duplicate "Ubuntu"
+profiles, for example).
+
+Additionally, each dynamic profile generator **must have a unique source guid**
+to associate with the profile. When a dynamic profile is generated, the source's
+guid will be added to the profile, to make sure the profile is correlated with
+the source it came from.
+
+We'll generate these dynamic profiles immediately after applying the defaults.
+When a generator runs, it'll be able to create unique profile GUIDs for each
+source it wants to generate a profile for. It'll attempt find any profiles that
+exist in the list of profiles with a matching GUID, and apply the settings to
+that profile if found, or it'll create a new profile to apply the settings to.
+
+After a dynamic profile generator runs, we will determine what new profiles need
+to be added to the user settings, so we can append those to the list of
+profiles. We'll store some sort of result indicating that we want a save
+operation to occur. After the rest of the deserializing is done, the app will
+then save the `profiles.json` file, including these new profiles.
+
+When we're serializing the settings, instead of comparing a dynamic profile to
+the default-constructed `Profile`, we'll compare it to the state of the
+`Profile` after the dynamic profile generator created it. It'd then only
+serialize settings that are different from the auto-generated version. It will
+also always make sure that the `guid` of the dynamic profile is included in the
+user settings file, as a point for the user to add customizations to the dynamic
+profile to. Additionally, we'll also make sure the `source` is always serialized
+as well, to keep the profile correlated with the generator that created it.
+
+We'll need to keep the state of these dynamically generated profiles around in
+memory during runtime to be able to ensure the only state we're serializing is
+that which is different from the initially generated dynamic profile.
+
+When the generator is run, and determines that a new profile has been added,
+we'll need to make sure to add the profile to the user's settings file. This
+will create an easy point for users to customize the dynamic profiles. When
+added to the user settings, all that will be added is the `name`, `guid`, and
+`source`.
+
+Additionally, a user might not want a dynamic profile generator to always run.
+They might want to keep their Azure connections visible in the list of profiles,

```

Comment:
I don't see any provisions above for the dynamic generators to delete profiles. Just add them if they don't exist. Which I thought was alright behavior.

But now you're talking about them disappearing when no longer a valid target and needing to disable the generator for it.

Couldn't we just instead have the behavior be "add if not exists" and only let the user remove them whenever they want?

Reply:
~~Ah shoot, that's one of the scenarios that was lost during #1321. Supposedly we can always go the `"hidden": true` route with them.~~
Or, I guess if (lets say) ubuntu is uninstalled. When the WSL generator runs, it won't create the ubuntu profile. When we get to the Ubunutu profile in the user's settings, it'll have a `source`, but we won't already have a profile with that `guid` and `source`. So we'll just ignore it, because whatever source for that profile doesn't want it anymore. Effectively, this will act like it was "deleted", though the artifacts still remain untouched in the user's json.

Reply:
#answered?

Reply:
Yeah ok.